### PR TITLE
Minor improvements on ClientConfig

### PR DESCRIPTION
**Pull request by gnunicorn**
- Documenting how to use a custom http client via `reqwest::ClientBuilder`
- use `user_aget`-API to make the code more consisce.

### CODE REVIEW

**Comment by gnunicorn** [comment]
odd... why was my main so out of date :thinking: ...

**Review by poljar — CHANGES_REQUESTED** [review]
I think the version should be gone from the user agent.

File: crates/matrix-sdk/src/http_client.rs [hunk 1]
```diff
@@ -247,17 +245,12 @@ pub(crate) fn client_with_config(config: &ClientConfig) -> Result<Client, HttpEr
...
-            Some(a) => a.clone(),
-            None => HeaderValue::from_str(&format!("matrix-rust-sdk {}", crate::VERSION))
-                .expect("Can't construct the version header"),
-        };
-
-        headers.insert(reqwest::header::USER_AGENT, user_agent);
+        let user_agent = config
+            .user_agent
+            .clone()
+            .unwrap_or_else(|| format!("matrix-rust-sdk {}", crate::VERSION));
```
[the remark this answers is not in the document]
Comment (gnunicorn) [reply crates/matrix-sdk/src/http_client.rs:251]: you think this is a bad thing? I think this is useful telemetry data, giving the homeserver some indication which platforms are running - and if you don't want it (as a dev) you _could_ always overwrite it. But if you don't like it, I care 2 out of 10, fine with removing it...

[the remark this answers is not in the document]
Comment (gnunicorn) [reply crates/matrix-sdk/src/http_client.rs:251]:
```suggestion
            .unwrap_or_else(|| "matrix-rust-sdk".to_owned());
```

**Review by poljar — APPROVED** [review]

**Pull request merged**